src/net_processing.cpp

```diff
@@ -897,7 +897,18 @@ class PeerManagerImpl final : public PeerManager
      *
      * Memory used: 1.3 MB
      */
-    CRollingBloomFilter m_recent_rejects GUARDED_BY(m_tx_download_mutex){120'000, 0.000'001};
+    std::unique_ptr<CRollingBloomFilter> m_recent_rejects GUARDED_BY(m_tx_download_mutex){nullptr};
```

Comment:
The recently merged `ActiveTipChange` (#30111) should to be updated too (`m_recent_rejects` and `m_recent_rejects_reconsiderable` are directly dereferenced there).

Reply:
Ha, good catch! I missed this because `reset()` is both a method on `unique_ptr` and `CRollingBloomFilter`.

This seems to be harmless and work out the way it should, i.e. `unique_ptr::reset()` will delete the current filter and on the next call to `RecentRejectsFilter()` it will be re-allocated (which afaict would have the same effect as resetting the filter with `CRollingBloomFilter::reset()`). But obviously, I should change it to `m_recent_rejects->reset()` to call `reset()` on the filter not the `unique_ptr`.

Reply:
Right, it shouldn't have compiled otherwise with `.reset()` instead of `->reset()`, the explanation makes sense! Funny (almost) bug!

Reply:
Renamed